src/backend/api/meals.js

```diff
@@ -1,12 +1,130 @@
+const { request, response } = require("express");
 const express = require("express");
+const Knex = require("knex");
 const router = express.Router();
 const knex = require("../database");
 
+//Returns all meals
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    console.log("in/api/meals");
+    // variable for all the querys I am using so that I can check if one of the requested queryes from the user doesn't match/exists
+    const querys = [
+      "maxPrice",
+      "title",
+      "createdAfter",
+      "limit",
+      "availableReservations",
+    ];
+    const requestQuerys = Object.keys(request.query);
+    const matchQuerys = [];
+    requestQuerys.some((item) => {
+      if (querys.includes(item)) {
+        matchQuerys.push(item);
+      }
+    });
+
+    let sortedMeals = await knex("meals");
```

Comment:
Generally, it's a good practice to try and use knex operations directly instead of first loading your data and then filtering them. That way, in case you have a very large amount of data in your database, you avoid having to load all of them.
An example would be:
```
const meals = await knex("meal").where("price", "<=", maxPrice)
```
Here though, you have so few meals it doesn't matter don't worry, just something to have in mind :)

Reply:
I'll change everything to use knex operations. 🤞